### PR TITLE
hide translations selector if only one translation available

### PR DESCRIPTION
**Pull request by jamesmcmurry**
Hide the left sidebar translations option if only one translation is defined in the site configuration

### CODE REVIEW

**Comment by ottok** [comment]
I have not tested this but idea looks good to me!

File: layouts/partials/sidebar/left.html [hunk 1]
```diff
@@ -77,17 +77,19 @@ <h2 class="site-description">{{ .Site.Params.sidebar.subtitle }}</h2>
...
-            {{ with .Site.Home.AllTranslations }}
-                <li id="i18n-switch">  
-                    {{ partial "helper/icon" "language" }}
-                    <select name="language" onchange="window.location.href = this.selectedOptions[0].value">
-                        {{ range . }}
-                            <option value="{{ .Permalink }}" {{ if eq .Language.Lang $currentLanguageCode }}selected{{ end }}>{{ .Language.LanguageName }}</option>
-                        {{ end }}
-                    </select>
-                </li>
+            {{ if ( compare.Gt .Site.Home.AllTranslations 1 ) }}
```
Comment (CaiJimmy) [inline layouts/partials/sidebar/left.html:80]: .Site.Home.AllTranslations` has type `Pages`. This fix works, but I think it's doing some kind of casting to compare a `Pages` to an `Int`.

I think it's better to avoid this kind of casting. We can use `.Site.Home.AllTranslations.Len` to get the length of the collection.

**Review by CaiJimmy — CHANGES_REQUESTED** [review]
Thanks for the PR! Please take a look at the comment.

**Comment by ottok** [comment]
I am seeing this bug on my site as well:

image

I confirm that the PR fixes it, but probably best to polish it following Jimmy's feedback.

**Comment by ottok** [comment]
Note that #925 is targeting the same section of code, so once either one is merged, the other one needs to be rebased.

**Comment by andreagrandi** [comment]
I was about to propose a similar pull request 😯 I was also seeing the same widget shown here https://github.com/CaiJimmy/hugo-theme-stack/pull/950#issuecomment-1962230394 even if I don't have any translated content in my whole website.

I did something different by the way (note: I'm using this theme as hugo module):

- I override `layouts/partials/sidebar/left.html` and changed that block in this way:

```html
            {{ if (default false .Site.Params.languageSelector.toggle) }}
                {{- $currentLanguageCode := .Language.Lang -}}
                {{ with .Site.Home.AllTranslations }}
                    <li id="i18n-switch">  
                        {{ partial "helper/icon" "language" }}
                        <select name="language" onchange="window.location.href = this.selectedOptions[0].value">
                            {{ range . }}
                                <option value="{{ .Permalink }}" {{ if eq .Language.Lang $currentLanguageCode }}selected{{ end }}>{{ .Language.LanguageName }}</option>
                            {{ end }}
                        </select>
                    </li>
                {{ end }}
            {{ end }}
```

basically I added this check `{{ if (default false .Site.Params.languageSelector.toggle) }}` at the beginning, so the widget is hidden, unless you explicitly add:
```yaml
languageSelector:
    toggle: true
```
to your params.

**Note**: this works, but I would hugely prefer if it was something fixed upstream (like this PR) so I don't have to maintain my own customisation.
